scope.go

```diff
@@ -239,14 +241,13 @@ func (s *scope) reportLoop(interval time.Duration) {
 }
 
 func (s *scope) reportLoopRun() {
-	// Need to hold a status lock to ensure not to report
-	// and flush after a close
-	s.status.RLock()
-	if s.status.closed {
-		s.status.RUnlock()
+	// don't flush after scope fully closed
+	if atomic.LoadInt32(&s.status.closed) == 2 {
 		return
 	}
 
+	s.status.flushes.Add(1)
```

Comment:
This may have to be done outside of reportLoopRun ie at the callers otherwise there's a race of wait being called when someone calls this and is at line 248.